library/src/main/java/dev/hichamboushaba/suspendactivityresult/ActivityProvider.kt

```diff
@@ -56,5 +56,7 @@ internal object ActivityProvider : Application.ActivityLifecycleCallbacks {
 
     override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {}
 
-    override fun onActivityDestroyed(activity: Activity) {}
+    override fun onActivityDestroyed(activity: Activity) {
+        _activityFlow.value = WeakReference(null)
+    }
```

Comment:
Please revert this, this would might break the logic on multi-activities apps, and this is not needed because:
1. The `WeakReference` will prevent leakage.
2. We check if the activity is still active [here](https://github.com/hichamboushaba/SuspendActivityResult/blob/05a26fe0b711f5f669e2c14eb92b8e670c228e13/library/src/main/java/dev/hichamboushaba/suspendactivityresult/ActivityProvider.kt#L22) before returning it.

Reply:
Thank you for answer. I didn't understand it before, because i have single activity in my project. I reverted it.